src/pages/HabitTracker/Habits.tsx

```diff
@@ -1,87 +1,256 @@
 import { useState, useEffect } from "react";
+import styled, { keyframes } from "styled-components";
 import { Modal, Button, Form } from "react-bootstrap";
-import styled from "styled-components";
 import _ from "lodash";
 import { PageBackground } from "../../components/styled/PageBackground";
 import { Sidebar } from "../../components/Sidebar";
 import { links } from "../../common/SidebarLinks";
 import { HabitTile } from "../../components/habittracker/HabitTile";
-import { AddHabitTile } from "../../components/habittracker/AddHabitTile";
+import { AddHabitButton } from "../../components/habittracker/AddHabitButton";
 import { StyledModal } from "../../components/styled/Modal";
+import { EditModeButton } from "../../components/habittracker/EditModeButton";
+import { DateRangePicker } from "../../components/habittracker/DateRangePicker";
+import { LogButton } from "../../components/habittracker/LogButton";
+import { EditButton } from "../../components/habittracker/EditButton";
+import { DeleteButton } from "../../components/habittracker/DeleteButton";
+import { RemoveLogButton } from "../../components/habittracker/RemoveLogButton";
+
+// Utility function to update habit data in state and localStorage
+const updateHabits = (
+  updatedHabits: Habit[],
+  setHabits: (habits: Habit[]) => void
+) => {
+  setHabits(updatedHabits);
+  localStorage.setItem("habits", JSON.stringify(updatedHabits));
+};
+
+// Utility function to update logging data in state and localStorage
+const updateLogging = (
+  updatedLogging: any,
+  setLogging: (logging: any) => void
+) => {
+  setLogging(updatedLogging);
+  localStorage.setItem("logging", JSON.stringify(updatedLogging));
+};
+
+// Utility function to reset habit form
+const resetHabitForm = (
+  setNewHabit: (habit: Habit) => void,
+  setOriginalHabit: (habit: Habit) => void
+) => {
+  setNewHabit({ name: "", count: 0 });
+  setOriginalHabit({ name: "", count: 0 });
+};
+
+// Utility function to manage logging data
+const manageLogging = (
+  habitName: string,
+  selectedDate: Date,
+  logging: any,
+  setLogging: (logging: any) => void,
+  action: "add" | "remove"
+) => {
+  const year = selectedDate.getFullYear();
+  const month = selectedDate
+    .toLocaleString("default", { month: "long" })
+    .toLowerCase();
+  const day = selectedDate.getDate();
+
+  const updatedLogging = { ...logging };
+
+  // Initialize logging structure if it doesn't exist
+  if (!updatedLogging[habitName]) {
+    updatedLogging[habitName] = {};
+  }
+  if (!updatedLogging[habitName][year]) {
+    updatedLogging[habitName][year] = {};
+  }
+  if (!updatedLogging[habitName][year][month]) {
+    updatedLogging[habitName][year][month] = {};
+  }
+  if (!updatedLogging[habitName][year][month][day]) {
+    updatedLogging[habitName][year][month][day] = 0;
+  }
+
+  // Add or remove log based on action
+  if (action === "add") {
+    updatedLogging[habitName][year][month][day] += 1;
+  } else if (
+    action === "remove" &&
```

Comment:
do not use hard-coded strings like `add` or `remove`. What if you make a mistake at any one place; it will break the code.
Use enums instead.